### PR TITLE
Support disabling the screen blank/lock

### LINKED ISSUE #4534: Allow apps to block screen locking/screensaver
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same feature, before opening a new one.
- [X] This issue only relates to a single feature. I will open new issues for any other features.

### Is your feature request related to a problem?

I was thinking of building a mobile monitoring app, and wanted the screen to remain on while the app is running.

I looked into it and there are simple native APIs for it.

On iOS you can call
```objective-c
[[NSOperationQueue mainQueue] addOperationWithBlock:^ {
    [UIApplication sharedApplication].idleTimerDisabled = YES;
}];
```
(Thanks Andy for finding this, I only found the Swift API)

On Android it's split into two options, either keep screen on or keep CPU running, not entirely sure we need the latter and the former doesn't require permissions:
```java
public class MainActivity extends Activity {
  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_main);
    getWindow().addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON);
  }
}
```
https://developer.android.com/develop/background-work/background-tasks/scheduling/wakelock
It looks like there's also a C version of this code but I could only find mention of it and not the API itself.

### Is it possible to construct a solution with the existing API?

_No response_

### Describe the solution you'd like to see.

I think probably just add to the mobile driver API a method that calls the native APIs, preferably something that can be set and reset, so you can keep screen on for specific long lasting tasks and let it lock otherwise

### PR DESCRIPTION
**Pull request by andydotxyz**
API proposed and implementation for iOS and Android.
Windows/macOS/Linux X11&Wayland could be another PR?

Fixes #4534

### Checklist:

- [ ] Tests included.
- [x] Lint and formatter run with no errors.
- [x] Tests all pass.
- [x] Public APIs match existing style and have Since: line.

### CODE REVIEW

**Comment by dweymouth** [comment]
Should this really be a desktop API as well? Isn't asking the screen not to sleep pretty much a mobile-only thing?

**Comment by andydotxyz** [comment]
I was going on what the linked issue described - the way that macOS, Windows and Linux allow apps to stop screensaver in the same way the mobile stops the screen switch off / lock.

**Comment by dweymouth** [comment]
> I was going on what the linked issue described - the way that macOS, Windows and Linux allow apps to stop screensaver in the same way the mobile stops the screen switch off / lock.
> 
> > to reflect that some OSes/DEs may refuse the request or not support it at all
> 
> I've not found one that can block this request yet.

Ah, OK, hadn't seen the latest comment activity on the ticket. I'd say if we can at least verify that we can support it on Windows, Mac, and X11, then we can go with the `Set...` naming as proposed in this PR. The actual implementation could wait until a follow-up PR

File: device.go [hunk 1]
```diff
@@ -31,6 +31,11 @@ type Device interface {
 	IsBrowser() bool
 	HasKeyboard() bool
 	SystemScaleForWindow(Window) float32
+
+	// SetDisableScreenBlanking allows an app to ask the device not to sleep/lock/blank displays
```
Comment (dweymouth) [inline device.go:35]: The `Device` interface comment will need updating since it no longer just "provides information" about the device

**Review by dweymouth — CHANGES_REQUESTED** [review]

File: device.go (hunk 1 above)
Comment (dweymouth) [reply device.go:35]: Alternatively, the API could be on the Driver type

Comment (andydotxyz) [reply device.go]: Good point - best not to change the intent of an interface ;)

**Review by dweymouth — APPROVED** [review]
I think this is good to go now, as long as we remember to add the desktop implementation before pulling it into a release

**Pull request merged**